client/src/components/Weather.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react';
+import './Weather.css';
+
+interface IWeatherProps {
+	sendLocation: (location: string) => void;
+}
+
+export const Weather = ({ sendLocation }: IWeatherProps) => {
+	const [location, setLocation] = useState<string>('');
+
+	const handleLocationInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setLocation(e.target.value);
+	};
+	useEffect(() => {
+		sendLocation(location);
```

Comment:
this is wrong, you're sending the location with every keystroke, like if i enter Los Angeles the server is receiving L, Lo, Los, Los A, etc

instead sendLocation should only be called when a button is pressed like the other panels

Reply:
changed to execute sendLocation only when the button is clicked.